src/lib/libpthread.js

```diff
@@ -251,7 +251,10 @@ var LibraryPThread = {
         } else if (cmd === 'spawnThread') {
           spawnThread(d);
         } else if (cmd === 'cleanupThread') {
-          cleanupThread(d.thread);
+          // Cleanup can potentially execute the queue mailbox and trigger an
+          // ExitStatus exception, so we must wrap cleanup with callUserCallback
+          // to handle the exception.
```

Comment:
How about "cleanupThread needs to be run via callUserCallback since it calls back into user code to free stuff.  Without this its possible the `unwind` or `ExitStatus` exception would escape here".